src/sock.c

```diff
@@ -281,8 +283,7 @@ static void read_conn(struct mg_connection *c) {
       if (!ioalloc(c, &c->rtls)) return;
       n = recv_raw(c, (char *) &c->rtls.buf[c->rtls.len],
                    c->rtls.size - c->rtls.len);
-      // MG_DEBUG(("%lu %ld", c->id, n));
-      if (n == MG_IO_ERR && mg_tls_pending(c) == 0 && c->rtls.len == 0) {
+      if (n == MG_IO_ERR && c->rtls.len == 0) {
         // Close only if we have fully drained both raw (rtls) and TLS buffers
```

Comment:
Should we also check for mg_tls_pending() as before ? Otherwise I think the comment is not accurate

Reply:
Good question - but I am unsure about that.
It looks like we might not need `mg_tls_pending()` at all.
TLS exchanges data by so-called "records", with "record length" specified at the beginning of the "record". so I expect TLS code to retrieve data from `c->rtls` by records. Meaning, if `c->rtls` is empty, there could not be anything pending. But that's a theory, that's why I did not remove `mg_tls_pending()` yet.
I suggest to give it some runway and see.